expression/builtin_like_test.go

```diff
@@ -36,14 +38,23 @@ func (s *testEvaluatorSuite) TestLike(c *C) {
 		{"aAb", `Aa%`, 0},
 		{"aAb", "aA_", 1},
 	}
-	for _, tt := range tests {
-		fc := funcs[ast.Like]
-		f, err := fc.getFunction(s.ctx, s.datumsToConstants(types.MakeDatums(tt.input, tt.pattern, 0)))
-		c.Assert(err, IsNil)
-		r, err := evalBuiltinFunc(f, chunk.Row{})
-		c.Assert(err, IsNil)
-		c.Assert(r, testutil.DatumEquals, types.NewDatum(tt.match))
+	wg := sync.WaitGroup{}
+	concurrency := 10
+	wg.Add(concurrency)
+	for i := 0; i < concurrency; i++ {
+		go func() {
+			defer wg.Done()
+			for _, tt := range tests {
+				fc := funcs[ast.Like]
+				f, err := fc.getFunction(s.ctx, s.datumsToConstants(types.MakeDatums(tt.input, tt.pattern, 0)))
```

Comment:
If an individual function signature is created in one goroutine, the bug may not be revealed.